### PR TITLE
Upgrade Halibut to net6.0 and net48

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Upgrade projects to net6.0 and net48

net45 and netstandard2.0 are no longer supported.
Tentacle targets net6.0 and net48 and Server targets net6.0
Aligning Halibut and removing usage of end of support framework versions

# How to review this PR

Quality :heavy_check_mark:

🟢  Tentacle: Upgrade Halibut => https://github.com/OctopusDeploy/OctopusTentacle/pull/463
🟢  Sever: Upgrade Tentacle => https://github.com/OctopusDeploy/OctopusDeploy/pull/17828

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.sln [hunk 1]
```diff
@@ -1,6 +1,7 @@
-﻿Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26228.4
+﻿
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.sln:1]: Every time I open Halibut in visual studio it changes the solution file 🙄 I can revert if needed

**Comment by nathanwoctopusdeploy** [comment]
[sc-47567]

File: source/Halibut.OctopusSample/Halibut.OctopusSample.csproj [hunk 2]
```diff
@@ -2,9 +2,10 @@
 
   <PropertyGroup>
     <VersionPrefix>0.0.0</VersionPrefix>
-    <TargetFramework>net45</TargetFramework>
+    <TargetFramework>net48</TargetFramework>
     <AssemblyName>Halibut.OctopusSample</AssemblyName>
     <OutputType>Exe</OutputType>
+    <UseWindowsForms>true</UseWindowsForms>
```
Comment (LukeButters) [inline source/Halibut.OctopusSample/Halibut.OctopusSample.csproj:8]: What is this about?

File: source/Halibut.CertificateGenerator/CertificateGenerator.cs [hunk 3]
```diff
@@ -80,7 +80,6 @@ static X509Certificate2 AttemptToGenerate(string fullName)
                    };
         }
 
-#if !__MonoCS__
```
[the remark this answers is not in the document]
Comment (LukeButters) [reply source/Halibut.CertificateGenerator/CertificateGenerator.cs:83]: We never compile with mono so this can't be used.

**Review by LukeButters — APPROVED** [review]

File: source/Halibut.OctopusSample/Halibut.OctopusSample.csproj (hunk 2 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.OctopusSample/Halibut.OctopusSample.csproj:8]: Had to add it to get the correct references for WinForms so the project would compile!!

Comment (nathanwoctopusdeploy) [reply source/Halibut.OctopusSample/Halibut.OctopusSample.csproj:8]: I'm not an expert here, google said this will fix it and it did 🤷

Comment (LukeButters) [reply source/Halibut.OctopusSample/Halibut.OctopusSample.csproj:8]: Oh i see I don't know more so if it works :D. What does chatgpt say :p

**Comment by nathanwoctopusdeploy** [comment]
Fixes: https://github.com/OctopusDeploy/Issues/issues/8229

**Pull request merged**